src/test/scala/com/vngrs/etl/loaders/JsonLoadSpec.scala

```diff
@@ -0,0 +1,73 @@
+package com.vngrs.etl.loaders
+
+import com.vngrs.etl.SparkSpec
+import com.vngrs.etl.utils.FileSystems
+import org.apache.spark.sql.{Row, SQLContext}
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+
+import scala.util.Random
+
+/**
+  * JSON Loader Specs
+  */
+// Following wart does not work with scalatest's intercept functionality
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class JsonLoadSpec extends SparkSpec {
+
+  "A json loader" should "load json data to files" in {
+    val path = getClass.getResource("/").getPath + s"$randomString"
```

Comment:
Instead of randomly writing to resource folder, we can create temporary files. `File.createTempFile`creates a temporary file with system call so the uniqueness of file will be ensured by the OS.